scheduler/server/load_based_sched_alg.go

```diff
@@ -0,0 +1,781 @@
+// is the difference between the min and max differences across the classes.  Eg: if 30% was the load target
+// for class A and class A is using 50% of the workers, A's pct difference is -20%.  If class B has a target
+// of 15% and is only using 5% of workers, B's pct difference is 10%.  If A and B are the only classes, the
+// pctSpread is 25% (5% - -20%).
+func (lbs *LoadBasedAlg) getCurrentPercentsSpread(totalWorkers int) int {
+	if len(lbs.jobClasses) < 2 {
+		return 0
+	}
+	minPct := 0
+	maxPct := 0
+	for _, className := range lbs.classByDescLoadPct {
+		jc := lbs.jobClasses[className]
+		currPct := int(math.Floor(float64(jc.origNumRunningTasks) * 100.0 / float64(totalWorkers)))
+		pctDiff := jc.origTargetLoadPct - currPct
+		if pctDiff < 0 || jc.numWaitingTasks > 0 { // only consider classes using loaned workers, or with waiting tasks
+			minPct = min(minPct, pctDiff)
+			maxPct = max(maxPct, pctDiff)
+		}
+	}
+
+	return maxPct - minPct
+}
+
+// getClassByDescLoadPct get a copy of the config's class by descending load pcts
+func (lbs *LoadBasedAlg) getClassByDescLoadPct() []string {
+	lbs.config.classLoadPercentsMu.RLock()
+	defer lbs.config.classLoadPercentsMu.RUnlock()
+	copy := []string{}
+	for _, v := range lbs.config.classByDescLoadPct {
+		copy = append(copy, v)
+	}
+	return copy
+}
+
+// getClassLoadPercents return a copy of the ClassLoadPercents converting to int32
+func (lbs *LoadBasedAlg) getClassLoadPercents() map[string]int32 {
+	lbs.config.classLoadPercentsMu.RLock()
+	defer lbs.config.classLoadPercentsMu.RUnlock()
+	copy := map[string]int32{}
+	for k, v := range lbs.config.classLoadPercents {
+		copy[k] = int32(v)
+	}
+	return copy
+}
+
+// LocalCopyClassLoadPercents return a copy of the ClassLoadPercents leaving as int
+func (lbs *LoadBasedAlg) LocalCopyClassLoadPercents() map[string]int {
+	lbs.config.classLoadPercentsMu.RLock()
+	defer lbs.config.classLoadPercentsMu.RUnlock()
+	copy := map[string]int{}
+	for k, v := range lbs.config.classLoadPercents {
+		copy[k] = v
+	}
+	return copy
+}
+
+// setClassLoadPercents set the scheduler's class load pcts with a copy of the input class load pcts
+func (lbs *LoadBasedAlg) setClassLoadPercents(classLoadPercents map[string]int32) {
+	lbs.config.classLoadPercentsMu.Lock()
+	defer lbs.config.classLoadPercentsMu.Unlock()
+
+	// build a list that orders the classes by descending pct.
+	keys := []string{}
+	for key := range classLoadPercents {
+		keys = append(keys, key)
+	}
+	sort.Slice(keys, func(i, j int) bool {
+		return classLoadPercents[keys[i]] > classLoadPercents[keys[j]]
+	})
+	lbs.config.classByDescLoadPct = keys
+
+	// set the load pcts - normalizing them if the don't add up to 100
+	lbs.config.classLoadPercents = map[string]int{}
+	pctTotal := 0
+	for _, val := range classLoadPercents {
+		pctTotal += int(val)
+	}
+	if pctTotal != 100 {
+		log.Errorf("LoadBalanced scheduling %%'s don't add up to 100, normalizing them")
```

Comment:
let's break normalization out into a helper so we can more easily test it

Reply:
done